samples/panther/sensor/src/main.c

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2016 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <device.h>
+#include <ipm.h>
+#include <ipm/ipm_quark_se.h>
+#include <misc/printk.h>
+#include <misc/util.h>
+#include <sensor.h>
+#include <zephyr.h>
+
+QUARK_SE_IPM_DEFINE(ess_ipm, 0, QUARK_SE_IPM_OUTBOUND);
+
+struct channel_info {
+	char               *dev_name;
+	struct device      *dev;
+	int                 chan;
+	struct sensor_value val;
+	int                 id;
+};
+
+static struct channel_info info[] = {
+
+#ifdef CONFIG_BME280
+	{"BME280", NULL, SENSOR_CHAN_TEMP    , {0, 0}, SENSOR_CHAN_TEMP},
+	{"BME280", NULL, SENSOR_CHAN_HUMIDITY, {0, 0}, SENSOR_CHAN_HUMIDITY},
+	{"BME280", NULL, SENSOR_CHAN_PRESS   , {0, 0}, SENSOR_CHAN_PRESS},
+#endif
+
+#ifdef CONFIG_ADXL362
+	{"ADXL362", NULL, SENSOR_CHAN_ACCEL_X, {0, 0}, SENSOR_CHAN_ACCEL_X},
+	{"ADXL362", NULL, SENSOR_CHAN_ACCEL_Y, {0, 0}, SENSOR_CHAN_ACCEL_Y},
+	{"ADXL362", NULL, SENSOR_CHAN_ACCEL_Z, {0, 0}, SENSOR_CHAN_ACCEL_Z},
+#endif
+
+#ifdef CONFIG_ADXRS290
+	{"ADXRS290", NULL, SENSOR_CHAN_GYRO_X, {0, 0}, SENSOR_CHAN_GYRO_X},
+	{"ADXRS290", NULL, SENSOR_CHAN_GYRO_Y, {0, 0}, SENSOR_CHAN_GYRO_Y},
+#endif
+
+#ifdef CONFIG_SI1153
+	{"SI1153", NULL, SENSOR_CHAN_GESTURE, {0, 0}, SENSOR_CHAN_GESTURE},
+#endif
+
+};
+
+void main(void)
+{
+	struct device *ipm;
+	struct device *old;
+	unsigned int i;
+	int rc;
+
+	ipm = device_get_binding("ess_ipm");
+	if (ipm == NULL) {
+		printk("Failed to get ESS IPM device\n");
+		return;
+	}
+
+	for (i = 0; i < ARRAY_SIZE(info); i++) {
+		info[i].dev = device_get_binding(info[i].dev_name);
+		if (info[i].dev == NULL) {
+			printk("Failed to get \"%s\" device\n", info[i].dev_name);
+		}
+	}
+
+	while (1) {
+
+		/* fetch sensor samples */
+		old = NULL;
+		for (i = 0; i < ARRAY_SIZE(info); i++) {
+			if (info[i].dev) {
```

Comment:
Better to do `if (!info[i].dev]) { continue; }` here. Avoids one indentation level.